modules/adkernelBidAdapter.js

```diff
@@ -1,310 +1,191 @@
-          utils._each(bidResp.seatbid[0].bid, (bid) => {
-            /* istanbul ignore else */
-            if (!bidFound && bid.impid === imp.id) {
-              bidFound = true;
-              callback(originalBids[imp.id], imp, bid);
-            }
-          });
-          if (!bidFound) {
-            callback(originalBids[imp.id], imp);
+export const spec = {
+
+  code: 'adkernel',
+  aliases: ['headbidding'],
+  supportedMediaTypes: [VIDEO],
+  isBidRequestValid: function(bidRequest) {
+    return 'params' in bidRequest && typeof bidRequest.params.host !== 'undefined' &&
+      'zoneId' in bidRequest.params && !isNaN(Number(bidRequest.params.zoneId));
+  },
+  buildRequests: function(bidRequests) {
+    let auctionId;
+    let dispatch = bidRequests.map(buildImp)
+      .reduce((acc, curr, index) => {
+        let bidRequest = bidRequests[index];
+        let zoneId = bidRequest.params.zoneId;
+        let host = bidRequest.params.host;
+        acc[host] = acc[host] || {};
+        acc[host][zoneId] = acc[host][zoneId] || [];
+        acc[host][zoneId].push(curr);
+        auctionId = bidRequest.bidderRequestId;
+        return acc;
+      }, {});
+    let requests = [];
+    Object.keys(dispatch).forEach(host => {
+      Object.keys(dispatch[host]).forEach(zoneId => {
+        const request = buildRtbRequest(dispatch[host][zoneId], auctionId);
+        requests.push({
+          method: 'GET',
+          url: `${window.location.protocol}//${host}/rtbg`,
+          data: {
+            zone: Number(zoneId),
+            ad_type: 'rtb',
+            v: VERSION,
+            r: JSON.stringify(request)
           }
         });
-      }, params, AJAX_REQ_PARAMS);
-    }
-
-    /**
-     * Builds complete rtb bid request
-     * @param imps collection of impressions
-     */
-    function buildRtbRequest(imps) {
-      return {
-        'id': utils.getUniqueIdentifierStr(),
-        'imp': imps,
-        'site': site,
-        'at': 1,
-        'device': {
-          'ip': 'caller',
-          'ua': 'caller'
-        }
-      };
-    }
-
-    /**
-     * Build ad-network specific endpoint url
-     */
-    function buildEndpointUrl(host) {
-      return `${window.location.protocol}//${host}/rtbg`;
+      });
+    });
+    return requests;
+  },
+  interpretResponse: function(serverResponse, request) {
+    /* istanbul ignore if */
+    if (!serverResponse.seatbid) {
+      return [];
     }
 
-    function buildRequestParams(zone, rtbReq) {
-      return {
-        'zone': encodeURIComponent(zone),
-        'ad_type': 'rtb',
-        'r': encodeURIComponent(JSON.stringify(rtbReq))
+    let rtbRequest = JSON.parse(request.data.r);
+    let rtbImps = rtbRequest.imp;
+    let rtbBids = serverResponse.seatbid
+      .map(seatbid => seatbid.bid)
+      .reduce((a, b) => a.concat(b), []);
+
+    return rtbBids.map(rtbBid => {
+      let imp = rtbImps.find(imp => imp.id === rtbBid.impid);
+      let prBid = {
+        bidderCode: spec.code,
```

Comment:
Our docs are bad... but you shouldn't send back the `bidderCode` here. The `bidderFactory` code does it automatically, and It breaks adapter aliasing if you do it yourself.